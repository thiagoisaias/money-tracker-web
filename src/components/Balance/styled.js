import styled from "styled-components";
import { devices } from "utils/devices";

export const Wrapper = styled.div`
  background-color: #fff;
  color: #484848;
  height: 100px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  box-shadow: 0.5px 1px 1px 1px #ddd;

  @media ${devices.small} {
    text-align: center;
    padding: 16px;
  }

  @media ${devices.mediumUp} {
    text-align: center;
    margin: 32px auto;
    border-radius: 2px;
    padding: 16px 32px;
  }
`;

export const Description = styled.p`
  margin-bottom: 8px;
`;

export const Value = styled.p`
  font-size: 24px;
  font-weight: 600;
`;
