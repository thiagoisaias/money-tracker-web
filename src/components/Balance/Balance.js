import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { devices } from "../../utils/devices";

const Container = styled.div`
  background-color: #fff;
  color: #484848;
  height: 100px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  box-shadow: 0.5px 1px 1px 1px #ddd;
  
  @media ${devices.small} {
    text-align: center;
    padding: 16px;
  }

  @media ${devices.mediumUp} {
    text-align: center;
    margin: 32px auto;
    border-radius: 2px;
    padding: 16px 32px;
  }
`;

const Description = styled.p`
  margin-bottom: 8px;
`;

const Value = styled.p`
  font-size: 24px;
  font-weight: 600;
`;

const mockProps = {
  currentBalance: 55127.32
};

const Balance = props => {
  const { currentBalance } = mockProps;
  return (
    <Container>
      <Description> {"Current Balance"}</Description>
      <Value>$ {currentBalance.toLocaleString()} </Value>
    </Container>
  );
};

Balance.propTypes = {
  currentBalance: PropTypes.number
};

export default Balance;
