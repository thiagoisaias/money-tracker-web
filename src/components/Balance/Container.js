import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Balance from "./Balance";

import { getOverallBalance } from "store/actions/users/users";

export class Container extends Component {
  render() {
    return <Balance {...this.props} />;
  }

  componentDidMount() {
    this.props.onGetOverallBalance();
  }
}

Container.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  overallBalance: PropTypes.number.isRequired
};

const mapStateToProps = state => ({
  isLoading: state.users.isLoading,
  overallBalance: state.users.overallBalance
});

const mapDispatchToProps = dispatch => ({
  onGetOverallBalance: () => {
    dispatch(getOverallBalance());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Container);
