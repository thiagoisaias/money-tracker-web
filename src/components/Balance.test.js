import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Balance from "./Balance";

describe("Balance", () => {
  const component = shallow(<Balance />);

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
