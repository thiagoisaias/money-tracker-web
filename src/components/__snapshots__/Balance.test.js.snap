// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance renders properly 1`] = `
<styled.div>
  <styled.p>
     Current Balance 
  </styled.p>
  <styled.p>
    R$ 
     
  </styled.p>
</styled.div>
`;
